sklearn/utils/testing.py

```diff
@@ -783,21 +783,35 @@ def __call__(self, *args, **kwargs):
 
 def _get_args(function, varargs=False):
```

Comment:
so how does this now differ from `sklearn.externals.funcsigs.signature`? Where does this avoid the pitfalls of that? Are there changes to funcsigs between Python 3.3 and 3.5 that we can use to improve `sklearn.externals.funcsigs`?

Reply:
There seem to have been changes to funcsigs since our copying it: https://pypi.python.org/pypi/funcsigs. Perhaps incorporating these will avoid the error you cite?

Reply:
I've just found out that `sklearn.externals.funcsigs.signature` only fails on buitin C function (after recursive call of `_get_parent_args`). I add try/catch exception to handle this and return empty array.